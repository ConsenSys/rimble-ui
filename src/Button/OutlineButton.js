import React from 'react';
import styled from 'styled-components';

import SolidButton from './SolidButton';

// outline
const StyledOutlineButton = styled(SolidButton)`
  & {
    color: var(--main-color);
    border: 1px solid;
    border-color: gray;
    background: transparent;
  }

  &&:hover {
    border-color: var(--main-color);
  }

  ${'' /* &&:hover::before {
    opacity: 0;
    background-color: var(--main-color);
  } */}

  &&:active::before {
    opacity: 0.1;
    background: var(--main-color);
  }
`;

const OutlineButton = ({ className, children, ...props }) => {
  return (
    <StyledOutlineButton className={className} {...props}>
      {children}
    </StyledOutlineButton>
  );
};

export default OutlineButton;
