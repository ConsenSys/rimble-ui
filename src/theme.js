import { tint, shade } from 'polished';

const success = '#28C081';
const danger = '#DC2C10';
const warning = '#FD9D28';
const info = '#36ADF1';

// theme.js
export default {
  fontSizes: [12, 14, 16, 20, 24, 32, 48, 64],
  fontWeights: [0, 300, 400, 600, 700],
  letterSpacings: [0, 1, 2, 4, 8],
  lineHeights: {
    solid: 1,
    title: 1.25,
    copy: 1.5,
  },
  fonts: {
    serif: 'athelas, georgia, times, serif',
    sansSerif: '"Source Sans Pro", -apple-system, sans-serif',
  },
  space: [0, 4, 8, 16, 32, 64, 128, 256],
  radii: ['0', '4px', '8px', '16px'],
  width: [0, 16, 32, 64, 128, 256],
  minWidths: [0, 16, 32, 64, 128, 256],
  maxWidths: [0, 16, 32, 64, 128, 256, 512, 768, 1024, 1536],
  heights: [0, 16, 32, 64, 128, 256],
  minHeights: [0, 16, 32, 64, 128, 256],
  maxHeights: [0, 16, 32, 64, 128, 256],
  borders: [0, '1px solid transparent'],
  borderWidths: ['0', '1px', '2px', '4px'],
  shadows: [
    '0',
    '0px 2px 4px rgba(0, 0, 0, 0.1)',
    '0 7px 14px rgba(50,50,93,.1)',
  ],
  opacity: {
    disabled: 0.4,
  },
  colors: {
    primary: '#4E3FCE',
    blue: '#007ce0',
    navy: '#004175',
    copyColor: '#3F3D4B',
    // black: '#000',
    black: '#000e1a',
    'near-black': '#111',
    'dark-gray': '#333',
    'mid-gray': '#555',
    // gray: ' #777',
    grey: '#CCC',
    silver: '#999',
    'light-silver': '#aaa',
    'moon-gray': '#ccc',
    'light-gray': '#eee',
    'near-white': '#f4f4f4',
    white: '#fff',
    transparent: 'transparent',
    blacks: [
      'rgba(0,0,0,.0125)',
      'rgba(0,0,0,.025)',
      'rgba(0,0,0,.05)',
      'rgba(0,0,0,.1)',
      'rgba(0,0,0,.2)',
      'rgba(0,0,0,.3)',
      'rgba(0,0,0,.4)',
      'rgba(0,0,0,.5)',
      'rgba(0,0,0,.6)',
      'rgba(0,0,0,.7)',
      'rgba(0,0,0,.8)',
      'rgba(0,0,0,.9)',
    ],
    whites: [
      'rgba(255,255,255,.0125)',
      'rgba(255,255,255,.025)',
      'rgba(255,255,255,.05)',
      'rgba(255,255,255,.1)',
      'rgba(255,255,255,.2)',
      'rgba(255,255,255,.3)',
      'rgba(255,255,255,.4)',
      'rgba(255,255,255,.5)',
      'rgba(255,255,255,.6)',
      'rgba(255,255,255,.7)',
      'rgba(255,255,255,.8)',
      'rgba(255,255,255,.9)',
    ],
    danger: danger,
    warning: warning,
    success: success,
    info: info,
  },
  zIndices: [0, 9, 99, 999, 9999],
  messageStyle: {
    base: {
      color: shade(0.4, '#AAA'),
      backgroundColor: tint(0.9, '#AAA'),
      borderColor: '#AAA',
    },
    success: {
      color: shade(0.4, success),
      backgroundColor: tint(0.9, success),
      borderColor: success,
    },
    warning: {
      color: shade(0.4, warning),
      backgroundColor: tint(0.9, warning),
      borderColor: warning,
    },
    danger: {
      color: shade(0.4, danger),
      backgroundColor: tint(0.9, danger),
      borderColor: danger,
    },
    info: {
      color: shade(0.4, info),
      backgroundColor: tint(0.9, info),
      borderColor: info,
    },
  },
};
