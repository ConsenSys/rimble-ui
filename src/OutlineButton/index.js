import React from 'react';

import styled from 'styled-components';
import { themeGet } from 'styled-system';
import { tint, shade } from 'polished';

import Button from '../Button';

import defaultTheme from '../theme';

const OutlineButton = styled(Button).attrs(props => ({
  primarycolor: themeGet('colors.primary', '#000')(props),
}))`
  & {
    background: transparent;
  }
  &&:hover {
    color: ${props => tint(0.2, props.primarycolor)};
    border-color: ${props => tint(0.2, props.primarycolor)};
    background-color: transparent;
  }
  &&:active {
    color: ${props => shade(0.2, props.primarycolor)};
    border-color: ${props => shade(0.2, props.primarycolor)};
    background-color: ${props => tint(0.9, props.primarycolor)};
  }
`;

OutlineButton.defaultProps = {
  theme: defaultTheme,
  color: 'primary',
  bg: 'transparent',
  fontWeight: 3,
  fontFamily: 'sansSerif',
  height: '48px',
  border: 1,
  borderColor: 'grey',
  px: 4,
  py: 0,
  borderRadius: 1,
};

OutlineButton.displayName = 'OutlineButton';

export default OutlineButton;
