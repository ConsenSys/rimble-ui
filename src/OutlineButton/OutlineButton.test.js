import OutlineButton from './'
import ThemeProvider from '../ThemeProvider'
import theme from '../theme'

describe('OutlineButton component sanity', () => {
  it('has name', () => {
    expect(OutlineButton.displayName).toBe('OutlineButton')
  })

  it('matches default snapshot', () => {
    const component = render( <OutlineButton /> )
    expect(component).toMatchSnapshot()
  })

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <OutlineButton />
      </ThemeProvider>
      )
    expect(component).toMatchSnapshot()
  })
})
