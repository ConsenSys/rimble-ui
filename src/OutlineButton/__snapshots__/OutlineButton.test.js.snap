// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton component sanity matches default snapshot 1`] = `
.c1 {
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 32px;
  padding-right: 32px;
  color: #4E3FCE;
  background-color: transparent;
  height: 48px;
  border: 1px solid transparent;
  border-color: #CCC;
  border-radius: 4px;
  border-color: #CCC;
  border-radius: 4px;
  text-align: left;
}

.c1 {
  box-sizing: border-box;
}

.c0 {
  font-weight: 600;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-size: 1rem;
  min-width: 3rem;
}

.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: 1;
  overflow: hidden;
  white-space: nowrap;
  -webkit-font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c0.c0:hover {
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  background-color: #5f52d2;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.2);
}

.c0:active {
  background-color: #4638b9;
  box-shadow: none;
}

.c0:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

.c0 > svg {
  display: block;
}

.c0 > svg:first-child {
  margin-left: -1rem;
  margin-right: .5rem;
}

.c0 > svg:last-child {
  margin-left: .5rem;
  margin-right: -1rem;
}

.c0 > svg:first-child:last-child {
  margin: 0;
}

.c0 {
  background: transparent;
}

.c0.c0:hover {
  color: #7165d7;
  border-color: #7165d7;
  background-color: transparent;
}

.c0.c0:active {
  color: #3e32a4;
  border-color: #3e32a4;
  background-color: #edebfa;
}

<button
  class="c0 c1"
  color="primary"
  font-family="sansSerif"
  font-weight="3"
  height="48px"
  type="button"
/>
`;

exports[`OutlineButton component sanity matches themed snapshot 1`] = `
.c2 {
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 32px;
  padding-right: 32px;
  color: #4E3FCE;
  background-color: transparent;
  height: 48px;
  border: 1px solid transparent;
  border-color: #CCC;
  border-radius: 4px;
  border-color: #CCC;
  border-radius: 4px;
  text-align: left;
}

.c2 {
  box-sizing: border-box;
}

.c1 {
  font-weight: 600;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  font-size: 1rem;
  min-width: 3rem;
}

.c1 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: 1;
  overflow: hidden;
  white-space: nowrap;
  -webkit-font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c1.c1:hover {
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  background-color: #5f52d2;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.2);
}

.c1:active {
  background-color: #4638b9;
  box-shadow: none;
}

.c1:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

.c1 > svg {
  display: block;
}

.c1 > svg:first-child {
  margin-left: -1rem;
  margin-right: .5rem;
}

.c1 > svg:last-child {
  margin-left: .5rem;
  margin-right: -1rem;
}

.c1 > svg:first-child:last-child {
  margin: 0;
}

.c1 {
  background: transparent;
}

.c1.c1:hover {
  color: #7165d7;
  border-color: #7165d7;
  background-color: transparent;
}

.c1.c1:active {
  color: #3e32a4;
  border-color: #3e32a4;
  background-color: #edebfa;
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <button
    class="c1 c2"
    color="primary"
    font-family="sansSerif"
    font-weight="3"
    height="48px"
    type="button"
  />
</div>
`;
